SupportFiles/ws-tools.sh

```diff
@@ -0,0 +1,161 @@
+chmod -R 755 /opt/idea-IC-183.5153.38
+ln -s /opt/idea-IC-183.5153.38/bin/idea.sh /usr/local/bin/idea
+
+cp /etc/cfn/tools/intellij/jetbrains-idea-ce.desktop /usr/share/applications/jetbrains-idea-ce.desktop
+cp /etc/cfn/tools/intellij/jetbrains-idea-ce.desktop /home/WorkstationUser/.local/share/applications/jetbrains-idea-ce.desktop
+chown WorkstationUser:WorkstationUser /home/WorkstationUser/.local/share/applications/jetbrains-idea-ce.desktop
+chmod 600 /home/WorkstationUser/.local/share/applications/jetbrains-idea-ce.desktop
+
+#Install emacs
+yum -y install emacs || err_exit "Failed to install emacs"
+
+#Install Gradle
+unzip -d /opt/ /etc/cfn/tools/gradle/gradle-5.1.1-bin.zip
+chmod -R 755 /opt/gradle-5.1.1
+ln -s /opt/gradle-5.1.1/bin/gradle /usr/local/bin/gradle
+
+# Install Maven
+yum -y install maven || err_exit "Failed to install maven"
+
+# Install git
+rpm -Uvh /etc/cfn/tools/git/endpoint-repo-1.7-1.x86_64.rpm
+yum -y install git  || err_exit "Failed to install git"
+yum -y install git-gui || err_exit "Failed to install git-gui"
+
+# Install ruby
+yum -y install ruby || err_exit "Failed to install ruby"
+
+# Install node.js
+tar -xvf  /etc/cfn/tools/nodejs/node-v11.6.0-linux-x64.tar.xz -C /opt
+chmod -R 755 /opt/node-v11.6.0-linux-x64
+ln -s /opt/node-v11.6.0-linux-x64/bin/node /usr/bin/node
+ln -s /opt/node-v11.6.0-linux-x64/bin/npm /usr/bin/npm
+
+# Install pycharm
+tar -xvf  /etc/cfn/tools/pycharm/pycharm-community-2018.3.3.tar.gz -C /opt
+chmod -R 755 /opt/pycharm-community-2018.3.3
+ln -s /opt/pycharm-community-2018.3.3/bin/pycharm.sh /usr/local/bin/pycharm 
+
+cp /etc/cfn/tools/pycharm/pycharm.desktop /usr/share/applications/pycharm.desktop
+cp /etc/cfn/tools/pycharm/pycharm.desktop /home/WorkstationUser/.local/share/applications/pycharm.desktop
+chown WorkstationUser:WorkstationUser /home/WorkstationUser/.local/share/applications/pycharm.desktop
+chmod 600 /home/WorkstationUser/.local/share/applications/pycharm.desktop     
+
+# Install asciidoctor tool chains 
+rpm -Uvh /etc/cfn/tools/asciidoctor/rubygem-asciidoctor-1.5.6.1-1.el7.noarch.rpm  
+yum -y install rubygem-asciidoctor || err_exit "Failed to install rubygem-asciidoctor"
+
+# Install Visual Studio Code
+rpm -Uvh /etc/cfn/tools/vscode/code-1.30.2-1546901769.el7.x86_64.rpm  
+yum -y install code || err_exit "Failed to install code"       
+
+# Install Mongo db Client – Mongo Shell                                          
+rpm -Uvh /etc/cfn/tools/mongo/mongodb-org-shell-4.0.5-1.el7.x86_64.rpm  
+sudo yum install -y mongodb-org-shell || err_exit "Failed to install mongodb-org-shell"       
+
+# Install MySQL and MySQL Workbench
+# The "proj" libary is required by mysql-workbench but missing from the epel-release-7-11.
+# Therefore, remove epel-release-7-11; and install epel-release-6-8.  We will remove epel-release-6-8
+# and re-install epel-release-7-11 later
+rpm -e epel-release-7-11.noarch
+rpm -Uvh /etc/cfn/tools/mysql/epel-release-6-8.noarch.rpm
```

Comment:
Never, ever, ever replace EL7 repo-defs with EL6 repo-defs. This can lead to major, major, major derangement down the road.

Overall, if the package is found in EL6's EPEL but not in EL7's EPEL, that typically means the target is now in the _vendor_ RPM channels: install from those (bonus: you'll likely get a newer, more-secure version by doing so).

Also: installing this kind of release package will cause the system to point to public repos: not good if the system cannot talk to the public Internet.

Reply:
Not done. Need to discuss with you further.

Reply:
The "proj" libary is required by mysql-workbench but missing from the epel-release-7-11.
Therefore, I need to remove epel-release-7-11 first; and then install epel-release-6-8 and proj.  At the end, I install  epel-release-7-11 again.